packages/composer/amazeelabs/silverback_gutenberg/src/Plugin/GraphQL/DataProducer/EditorBlockMedia.php

```diff
@@ -26,31 +33,122 @@
  *   }
  * )
  */
-class EditorBlockMedia extends DataProducerPluginBase {
-  public function resolve(
-    $block,
-    FieldContext $field
+class EditorBlockMedia extends DataProducerPluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity repository service.
+   *
+   * @var \Drupal\Core\Entity\EntityRepositoryInterface
+   */
+  protected $entityRepository;
+
+  /**
+   * The entity buffer service.
+   *
+   * @var \Drupal\graphql\GraphQL\Buffers\EntityBuffer
+   */
+  protected $entityBuffer;
+
+  /**
+   * {@inheritdoc}
+   *
+   * @codeCoverageIgnore
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('entity.repository'),
+      $container->get('graphql.buffer.entity')
+    );
+  }
+
+  /**
+   * EntityLoad constructor.
+   *
+   * @param array $configuration
+   *   The plugin configuration array.
+   * @param string $pluginId
+   *   The plugin id.
+   * @param array $pluginDefinition
+   *   The plugin definition array.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface $entityRepository
+   *   The entity repository service.
+   * @param \Drupal\graphql\GraphQL\Buffers\EntityBuffer $entityBuffer
+   *   The entity buffer service.
+   *
+   * @codeCoverageIgnore
+   */
+  public function __construct(
+    array $configuration,
+          $pluginId,
+    array $pluginDefinition,
+    EntityTypeManagerInterface $entityTypeManager,
+    EntityRepositoryInterface $entityRepository,
+    EntityBuffer $entityBuffer
   ) {
-    $mediaId = $block['attrs']['mediaEntityIds'][0] ?? NULL;
+    parent::__construct($configuration, $pluginId, $pluginDefinition);
+    $this->entityTypeManager = $entityTypeManager;
+    $this->entityRepository = $entityRepository;
+    $this->entityBuffer = $entityBuffer;
+  }
 
-    if (!$mediaId) {
-      return NULL;
-    }
-    $media = Media::load($mediaId);
-    // Try to get the embedded item in the current language or use whatever
-    // we get as a fallback.
-    if (!$media) {
-      \Drupal::logger('graphql_gutenberg')->notice("Cannot load media by ID '{$mediaId}'");
+  public function resolve($block, FieldContext $context): ?Deferred {
+    $id = $block['attrs']['mediaEntityIds'][0] ?? NULL;
+
+    if (!$id) {
       return NULL;
```

Comment:
From PR description:
> Turns out that `$builder->compose` only aborts on `NULL` values when they are emitted by a `Deferred` result.

But here `NULL` is still not wrapped with `Deferred`. Should it be wrapped as well?